src/components/TodoList/TodoList.tsx

```diff
@@ -1,208 +1,66 @@
-/* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { useAppSelector } from '../../app/hooks';
+import { TodoInfo } from '../TodoInfo/TodoInfo';
 
 export const TodoList: React.FC = () => {
+  const todos = useAppSelector(store => store.todos);
+  const filteration = useAppSelector(store => store.filter);
+  const [filteredTodos, setFilteredTodos] = useState(todos);
+  const [downloadingTodos, setDownloadingTodos] = useState(false);
+
+  const filtrationHandler = () => {
+    switch (filteration.status) {
+      case 'completed':
+        return todos.filter(todo => todo.completed)
+          .filter(todo => todo.title.includes(filteration.query));
+      case 'active':
+        return todos.filter(todo => !todo.completed)
+          .filter(todo => todo.title.includes(filteration.query));
+      default:
+        return todos.filter(todo => todo.title.includes(filteration.query));
+    }
+  };
+
+  useEffect(() => {
+    setFilteredTodos(filtrationHandler());
+  }, [filteration, todos]);
+
+  useEffect(() => {
+    setTimeout(() => {
+      setDownloadingTodos(true);
+    }, 500);
```

Comment:
Clear the timeout on the unmount, please. Made research on why we need this. You can read [here](https://bobbyhadz.com/blog/react-cleartimeout) for example